homeassistant/components/yolink/coordinator.py

```diff
@@ -35,77 +29,17 @@ def __init__(
         super().__init__(
             hass, _LOGGER, name=DOMAIN, update_interval=timedelta(minutes=30)
         )
-        self._client = yl_client
-        self._mqtt_client = yl_mqtt_client
-        self.yl_devices: list[YoLinkDevice] = []
-        self.data = {}
+        self.device = device
 
-    def on_message_callback(self, message: tuple[str, BRDP]):
-        """On message callback."""
-        data = message[1]
-        if data.event is None:
-            return
-        event_param = data.event.split(".")
-        event_type = event_param[len(event_param) - 1]
-        if event_type not in (
-            "Report",
-            "Alert",
-            "StatusChange",
-            "getState",
-        ):
-            return
-        resolved_state = data.data
-        if resolved_state is None:
-            return
-        self.data[message[0]] = resolved_state
-        self.async_set_updated_data(self.data)
-
-    async def init_coordinator(self):
-        """Init coordinator."""
-        try:
-            async with async_timeout.timeout(10):
-                home_info = await self._client.get_general_info()
-                await self._mqtt_client.init_home_connection(
-                    home_info.data["id"], self.on_message_callback
-                )
-            async with async_timeout.timeout(10):
-                device_response = await self._client.get_auth_devices()
-
-        except YoLinkAuthFailError as yl_auth_err:
-            raise ConfigEntryAuthFailed from yl_auth_err
-
-        except (YoLinkClientError, asyncio.TimeoutError) as err:
-            raise ConfigEntryNotReady from err
-
-        yl_devices: list[YoLinkDevice] = []
-
-        for device_info in device_response.data[ATTR_DEVICE]:
-            yl_devices.append(YoLinkDevice(device_info, self._client))
-
-        self.yl_devices = yl_devices
-
-    async def fetch_device_state(self, device: YoLinkDevice):
-        """Fetch Device State."""
+    async def _async_update_data(self) -> dict:
+        """Fetch device state."""
         try:
             async with async_timeout.timeout(10):
-                device_state_resp = await device.fetch_state_with_api()
+                device_state_resp = await self.device.fetch_state_with_api()
                 if ATTR_DEVICE_STATE in device_state_resp.data:
```

Comment:
Please only wrap the line that can raise in the try... except block. We can move these lines to the end of the function body.